packages/main/src/plugin/util/port.spec.ts

```diff
@@ -0,0 +1,36 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { expect, test } from 'vitest';
+import * as port from './port.js';
+import * as net from 'net';
+
+test('return port range starting with port 9000', async () => {
+  const range = await port.getFreePortRange(3);
+
+  expect(range).toBe('9000-9002');
+});
+
+test('return port range starting with port 9001 if 9000 is busy', async () => {
+  const server = net.createServer();
+  server.listen(9000);
+  const range = await port.getFreePortRange(3);
+
+  expect(range).toBe('9001-9003');
+  server.close();
+});
```

Comment:
how will it work if I have something running on my side on the port 9000 ? (I know I saw some apps listening on 9000 port)

should the test should pickup a free port/range before ?

Reply:
you're right, so i guess it only makes sense to check if the range is valid and the start/end values are valid numbers

Reply:
for the first one, yes probably you ask for a range and then create server on it then it ensures it gave you free ports

And for the second probably you ask for a range, get the ports then use the first port and ask again and see that it's shifted

Reply:
@benoitf updated